include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -514,7 +514,7 @@ struct __result_and_scratch_storage_base
     __get_data(sycl::event, std::size_t* __p_buf) const = 0;
 };
 
-template <typename _T>
+template <typename _T, std::size_t _NResults>
```

Comment:
Can we default this to `1` and get rid of a significant number of these changes?
(perhaps it is not worth losing the clarity when looking at the type, but I think it might be.)

Reply:
Of course, it's implementable.
But as far as I remembering initially we likes mandatory specify the amount of results and the amount of scratch-data.
But I will do.

Reply:
I'm OK with either option really.  Do what you think is best here.

Reply:
Done.